src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala

```diff
@@ -31,4 +50,207 @@ class PodsResource @Inject() (
+
+      val pod = PodDefinition(withDefaults(podDef, podDefaults)).withCanonizedIds()
+
+      withAuthorization(CreateRunSpec, pod) {
+        val (createdPod, deploymentId) = podSystem.create(pod, force)
+        Events.maybePost(PodEvent(req.getRemoteAddr, req.getRequestURI, PodEvent.Created))
+
+        val builder = Response
+          .created(new URI(createdPod.id.toString))
+          .entity(marshalJson(createdPod.asPodDef))
+
+        deploymentId.foreach(id => builder.header(DEPLOYMENT_ID_HEADER, id))
+        builder.build()
+      }
+    }(createPodValidator)
+  }
+
+  @GET @Timed
+  def findAll(@Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    val authSelector = Selector { pod =>
+      isAuthorized(ViewRunSpec, pod)
+    }
+    val pods: Iterable[PodDefinition] = podSystem.findAll(authSelector)
+    ok(marshalJson(pods.map(_.asPodDef)))
+  }
+
+  @GET @Timed @Path("""{id:.+}""")
+  def find(
+    @PathParam("id") id: String,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(id) { _ =>
+      val pod: PodDefinition = podSystem.find(id)
+      withAuthorization(ViewRunSpec, pod) {
+        ok(marshalJson(pod.asPodDef))
+      }
+    }(PodsValidation.idValidator)
+  }
+
+  @PUT @Timed @Path("""{id:.+}""")
+  def update(
+    @PathParam("id") id: String,
+    body: Array[Byte],
+    @DefaultValue("false")@QueryParam("force") force: Boolean,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(unmarshalJson[PodDef](decodeBytes(body, req))) { podDef =>
+
+      // podDef.id has just been validated, requiring equality for id validates it
+      require(id == podDef.id)
+      val pod = PodDefinition(withDefaults(podDef, podDefaults)).withCanonizedIds()
+
+      withAuthorization(UpdateRunSpec, pod) {
+
+        val (updatedPod, deploymentId) = podSystem.update(pod, force)
+        Events.maybePost(PodEvent(req.getRemoteAddr, req.getRequestURI, PodEvent.Updated))
+
+        val builder = Response
+          .ok(new URI(updatedPod.id.toString))
+          .entity(marshalJson(updatedPod.asPodDef))
+
+        deploymentId.foreach(id => builder.header(DEPLOYMENT_ID_HEADER, id))
+        builder.build()
+      }
+    }(createPodValidator and updatePodValidator)
+  }
+
+  @DELETE @Timed @Path("""{id:.+}""")
+  def remove(
+    @PathParam("id") id: String,
+    @DefaultValue("false")@QueryParam("force") force: Boolean,
+    @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
+
+    withValid(id) { _ =>
+      val pod: PodDefinition = podSystem.find(id)
+      withAuthorization(DeleteRunSpec, pod) {
+
+        val (deletedPod, deploymentId) = podSystem.delete(id, force)
+        Events.maybePost(PodEvent(req.getRemoteAddr, req.getRequestURI, PodEvent.Deleted))
+
+        val builder = Response
+          .ok(new URI(deletedPod.id.toString))
```

Comment:
Good point. I don't see where it **hurts** to provide this information ("this is the pod that was deleted"). But if it's confusing then we should remove